halyard-deploy/src/main/resources/kubernetes/manifests/service.yml

```diff
@@ -10,7 +10,11 @@ spec:
   selector:
     app: spin
     cluster: spin-{{ name }}
+  type: {{ type }}
   ports:
   - protocol: TCP
     port: {{ port }}
     targetPort: {{ port }}
+  {% if nodePort != null %}
```

Comment:
you probably also want to check to see if the type is `NodePort` and only set this if both those characteristics are true.